airbyte-cdk/java/airbyte-cdk/core/src/testFixtures/kotlin/io/airbyte/cdk/extensions/LoggingInvocationInterceptor.kt

```diff
@@ -0,0 +1,335 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+package io.airbyte.cdk.extensions
+
+import java.lang.reflect.*
+import java.time.Duration
+import java.time.Instant
+import java.time.format.DateTimeParseException
+import java.util.*
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.TimeoutException
+import java.util.regex.Pattern
+import kotlin.concurrent.Volatile
+import org.apache.commons.lang3.StringUtils
+import org.apache.commons.lang3.exception.ExceptionUtils
+import org.apache.commons.lang3.time.DurationFormatUtils
+import org.junit.jupiter.api.Timeout
+import org.junit.jupiter.api.extension.DynamicTestInvocationContext
+import org.junit.jupiter.api.extension.ExtensionContext
+import org.junit.jupiter.api.extension.InvocationInterceptor
+import org.junit.jupiter.api.extension.ReflectiveInvocationContext
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+/**
+ * By default, junit only output logs to the console, and nothing makes it into log4j logs. This
+ * class fixes that by using the interceptor facility to print progress and timing information. This
+ * allows us to have junit loglines in our test logs. This is instanciated via
+ * [Java's
+ * ServiceLoader](https://docs.oracle.com/javase%2F9%2Fdocs%2Fapi%2F%2F/java/util/ServiceLoader.html)
+ * The declaration can be found in
+ * resources/META-INF/services/org.junit.jupiter.api.extension.Extension
+ */
+class LoggingInvocationInterceptor : InvocationInterceptor {
+    private class LoggingInvocationInterceptorHandler : InvocationHandler {
+        @Throws(Throwable::class)
+        override fun invoke(proxy: Any?, method: Method?, args: Array<Any?>?): Any? {
+            if (
+                LoggingInvocationInterceptor::class
+                    .java
+                    .getDeclaredMethod(
+                        method!!.name,
+                        InvocationInterceptor.Invocation::class.java,
+                        ReflectiveInvocationContext::class.java,
+                        ExtensionContext::class.java
+                    ) == null
+            ) {
+                LOGGER!!.error(
```

Comment:
We should use [kotlin-logging](https://github.com/oshai/kotlin-logging) over the direct SLF4J logger.  I would recommend that we do that as part of this change, though it could be done as a follow-on pass.